sdk/storage/AzureStorageBlob/Source/DataStructures/StorageStreamUploader.swift

```
@@ -0,0 +1,540 @@
+    ///   - url: The source URL of the blob object.
+    ///   - blockId: A unique block identifer.
+    ///   - startRange: The start point, in bytes, of the upload request.
+    ///   - endRange: The end point, in bytes, of the upload request.
+    ///   - options: An `UploadBlobOptions` object with which to control the upload.
+    public init(
+        blob: String,
+        container: String,
+        blockId: String,
+        client: StorageBlobClient,
+        url: URL,
+        startRange: Int,
+        endRange: Int,
+        options: UploadBlobOptions
+    ) {
+        self.blockId = blockId
+        self.blobName = blob
+        self.containerName = container
+        self.client = client
+        self.options = options
+        self.uploadSource = url
+        self.startRange = startRange
+        self.endRange = endRange
+    }
+
+    // MARK: Public Methods
+
+    /// Begin the upload process.
+    /// - Parameters:
+    ///   - requestId: Unique request ID (GUID) for the operation.
+    ///   - completion: A completion handler that forwards the downloaded data.
+    public func upload(
+        requestId: String? = nil,
+        transactionalContentMd5: Data? = nil,
+        transactionalContentCrc64: Data? = nil,
+        then completion: @escaping HTTPResultHandler<Data>
+    ) {
+        // Construct URL
+        let urlTemplate = "/{container}/{blob}"
+        let pathParams = [
+            "container": containerName,
+            "blob": blobName
+        ]
+        let url = client.url(forTemplate: urlTemplate, withKwargs: pathParams)
+        let chunkSize = endRange - startRange
+        var buffer = Data(capacity: chunkSize)
+        do {
+            guard let fileHandle = try? FileHandle(forReadingFrom: uploadSource) else {
+                throw AzureError.general("Unable to open for reading: \(uploadSource)")
+            }
+            if #available(iOS 13.0, *) {
+                try fileHandle.seek(toOffset: UInt64(startRange))
```

Is there not an equivalent of `try?` for methods that may throw but return void? Odd that we have to encase the thing in a `do... catch` block just because the method doesn't produce a return value.

---

Yeah, let me take a look :)

---

I took a look at this section and it's wrapped in a do/catch to catch various file-related errors that could occur. I updated the logic and removed the TODO. It now calls a `completion(.failure(error))` to ensure that any file errors are surfaced to the developer instead of being swallowed.